pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -37,24 +37,26 @@ func NewOpenShiftClusterDynamicValidator(
 	oc *api.OpenShiftCluster,
 	subscriptionDoc *api.SubscriptionDocument,
 	fpAuthorizer autorest.Authorizer,
-) OpenShiftClusterDynamicValidator {
-	return &openShiftClusterDynamicValidator{
+) *OpenShiftClusterDynamicValidator {
+	return &OpenShiftClusterDynamicValidator{
 		log: log,
 		env: env,
 
 		oc:              oc,
 		subscriptionDoc: subscriptionDoc,
 		fpAuthorizer:    fpAuthorizer,
+		byoNSG:          feature.IsRegisteredForFeature(subscriptionDoc.Subscription.Properties, api.FeatureFlagBYONsg),
```

Comment:
moving the feature flag to the oc document and resolving it early will clean this up.